src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -122,4 +149,30 @@ String servicePointUserFilter(String s) {
           return tl.perform(attributes, headers, vertxContext, superRecordsLoaded);
         });
   }
+
+  private Future<Void> runJavaMigrations(TenantAttributes ta, Context context,
+    Map<String, String> okapiHeaders) {
+
+    log.info("About to start java migrations...");
+
+    List<BaseMigrationService> javaMigrations = List.of(
+      new ItemShelvingOrderMigrationService(context, okapiHeaders));
+
+    @SuppressWarnings("rawtypes")
+    List<Future> startedMigrations = javaMigrations.stream()
+      .filter(javaMigration -> javaMigration.shouldExecuteMigration(ta))
+      .peek(migration -> log.info(
+        "Following migration is to be executed [migration={}]", migration))
+      .map(BaseMigrationService::runMigration)
+      .collect(Collectors.toList());
+
+    return CompositeFuture.all(startedMigrations).<Void>map(result -> null)
+      .onSuccess(notUsed -> log.info("Java migrations has been completed"))
+      .onFailure(error -> log.error("Some java migrations failed", error));
```

Comment:
Does this mean that if the upgrade of these records fail, the module upgrade still succeeds?

Reply:
No, the error is still propagated to the caller, see `shouldStopProcessingWhenCannotConvertItem` test.